crates/weaver_resolver/src/body.rs

```diff
@@ -0,0 +1,88 @@
+
+// SPDX-License-Identifier: Apache-2.0
+
+//! Functions to resolve a semantic convention body.
+
+use weaver_resolved_schema::body::{Body, BodyField};
+use weaver_semconv::body::{BodyFieldSpec, BodySpec};
+
+use crate::attribute::AttributeCatalog;
+
+/// Resolve a body specification into a resolved body.
+pub fn resolve_body(
+    body: &BodySpec,
+    attr_catalog: &mut AttributeCatalog
+) -> Option<Body> {
+    match body {
+        BodySpec::Fields { fields } => {
+            let fields = fields.iter().map(|field| resolve_body_field(field, attr_catalog)).collect();
+            Some(Body {
+                fields: Some(fields),
+                value: None,
+            })
+        },
+        BodySpec::Value { value: _ } => {
+            // Add as a placeholder for now of where to resolve the value.
+            todo!("Implement value type for body.")
+        }
+    }
+}
+
+/// Resolve a body field specification into a resolved body field.
+pub fn resolve_body_field(
+    field: &BodyFieldSpec,
+    attr_catalog: &mut AttributeCatalog
+) -> BodyField {
+    match field {
+        BodyFieldSpec::Ref {
+            r#ref,
+            alias,
+            brief,
+            examples,
+            requirement_level,
+            note,
+            stability,
+            deprecated,
+        } => {
+            let attr_ref = attr_catalog.get_attribute_ref(r#ref);
+
+            let root_attr = attr_ref.attr.attribute.clone();
+            BodyField {
+                name: field.id(),
+                r#attr: Some(attr_ref.attr_ref),
+                alias: alias.clone(),
+                r#type: root_attr.r#type.clone(),
+                brief: if brief.is_some() { brief.clone().unwrap_or_default() } else { root_attr.brief.to_owned() },
+                examples: if examples.is_some() { examples.clone() } else { root_attr.examples.to_owned() },
+                requirement_level: if requirement_level.is_some() { requirement_level.as_ref().unwrap().clone() } else { root_attr.requirement_level.to_owned() },
+                note: if note.is_some() { note.clone().unwrap_or_default() } else { root_attr.note.to_owned() },
+                stability: if stability.is_some() { stability.clone() } else { root_attr.stability.to_owned() },
+                deprecated: if deprecated.is_some() { deprecated.clone() } else { root_attr.deprecated.to_owned() },
+            }
+        },
+        BodyFieldSpec::Id {
+            id,
+            r#type,
+            brief,
+            examples,
+            requirement_level,
+            note,
+            stability,
+            deprecated,
+        } => {
+            BodyField {
```

Comment:
Once you have a FieldCatalog, you probably need to feed it with body field definition already resolved (i.e. field with id).

Reply:
I don't think we are going to need the FieldCatalog, based on my comment above as the fields are not and should not be shared.